metadata-ingestion/src/datahub/ingestion/source/data_lake/__init__.py

```diff
@@ -538,14 +546,28 @@ def get_workunits_s3(self) -> Iterable[MetadataWorkUnit]:
 
             base_obj_path = f"{obj.bucket_name}/{obj.key}"
 
-            base_obj_paths.append(base_obj_path)
-
-        for aws_file in sorted(base_obj_paths):
-
-            relative_path = "./" + aws_file[len(plain_base_path) :]
+            properties = {
+                "owner": str(obj.owner) if obj.owner else "",
+                "e_tag": str(obj.e_tag) if obj.e_tag else "",
+                "last_modified": str(obj.last_modified) if obj.last_modified else "",
+                "size": str(obj.size) if obj.size else "",
+                "storage_class": str(obj.storage_class) if obj.storage_class else "",
+                "service_name": str(obj.meta.service_name)
+                if obj.meta and obj.meta.service_name
+                else "",
+            }
+            logger.debug(f"Adding file {base_obj_path} for ingestion")
+            base_obj_paths.append((base_obj_path, properties))
+
+        for aws_file in sorted(base_obj_paths, key=lambda a: a[0]):
+            path = aws_file[0]
+            properties = aws_file[1]
+            relative_path = "./" + path[len(plain_base_path) :]
```

Comment:
nit: prefer the slice function over the [len(plain_base_path) :] notation in future.